src/DynamoApplications/StartupUtils.cs

```diff
@@ -220,7 +233,21 @@ public static DynamoModel MakeModel(bool CLImode, string asmPath = "", string ho
         {
             PreloadASM(asmPath, out string geometryFactoryPath, out string preloaderLocation);
             return StartDynamoWithDefaultConfig(CLImode, geometryFactoryPath, preloaderLocation, hostName);
+        }
 
+        /// <summary>
+        /// Use this overload to construct a DynamoModel when the location of ASM to use is known and host name is known.
+        /// </summary>
+        /// <param name="CLImode">CLI mode starts the model in test mode and uses a separate path resolver.</param>
+        /// <param name="asmPath">Path to directory containing geometry library binaries</param>
+        /// <param name="hostName">Dynamo variation identified by host.</param>
+        /// <param name="parentId">Dynamo host parent id for analytics purpose</param>
+        /// <param name="sessionId">Dynamo host session id for analytics purpose</param>
+        /// <returns></returns>
+        public static DynamoModel MakeModel(bool CLImode, string asmPath = "", string hostName = "", string parentId = "", string sessionId = "")
```

Comment:
maybe its time to put these analytics related properties in a separate struct ? might.. make it easier to pass around..

Reply:
yeah, if this is easy @QilongTang seems like a good idea to wrap them up into some options bag, we won't need to keep adding new constructors for MakeModel or worrying about binary compatibility with overloads.

Reply:
I have wrapped the strings into a struct as part of this PR